sql/moz-fx-data-shared-prod/firefox_desktop_derived/snippets_users_daily_v2/query.sql

```diff
@@ -0,0 +1,42 @@
+WITH windowed AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    -- For all dimensions, we use the mode of observed values in the day.
+    udf.mode_last(ARRAY_AGG(release_channel) OVER w1) AS release_channel,
+    udf.mode_last(ARRAY_AGG(locale) OVER w1) AS locale,
+    udf.mode_last(ARRAY_AGG(metadata.geo.country) OVER w1) AS country,
+    udf.mode_last(ARRAY_AGG(version) OVER w1) AS version
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_derived.snippets_v2`
+  WHERE
+    -- Reprocess all dates by running this query with --parameter=submission_date:DATE:NULL
+    (@submission_date IS NULL OR @submission_date = DATE(submission_timestamp))
+  WINDOW
+    w1 AS (
+      PARTITION BY
+        client_id,
+        DATE(submission_timestamp)
+      ORDER BY
+        submission_timestamp
+      ROWS BETWEEN
+        UNBOUNDED PRECEDING
+        AND UNBOUNDED FOLLOWING
+    ),
+    -- We must provide a modified window for ROW_NUMBER which cannot accept a frame clause.
+    w1_unframed AS (
+      PARTITION BY
+        client_id,
+        DATE(submission_timestamp)
+      ORDER BY
+        submission_timestamp
+    )
+)
+--
+SELECT
```

Comment:
Might be able to avoid this extra step by using `QUALIFY` inside `windowed` to achieve the same result.

Reply:
No worries, was just mentioning a possible alternative. I'm also happy to leave it as is.